### PR TITLE
Issue 298 Language switch

### LINKED ISSUE #298: Language switch
UI control in the navbar where the user gets to change the language in real time.

### LINKED ISSUE #301: Initialize UI language as user's browser language
This is current behaviour of DSpace to be reproduced. I dipped into this, based on code available from the ngx-translate example at https://stackblitz.com/github/ngx-translate/example, but couldn't get this to work yet in DSpace

### PR DESCRIPTION
**Pull request by bram-atmire**
This PR closes #304
This PR closes #298
This PR closes #301

Hopefully final rebased and squashed version of the language switch feature, including successful tests.

### CODE REVIEW

**Comment by bram-atmire** [comment]
See https://github.com/DSpace/dspace-angular/pull/308 for the history and dialogue that lead to this final version.

**Comment by bram-atmire** [comment]
Spoke too soon, still have a number of lint style issues to address, apologies.

**Comment by bram-atmire** [comment]
Checks are passing now - I have currently no more changes pending here.

File: src/config/lang-config.interface.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Config } from './config.interface';
+
+export interface LangConfig extends Config {
```
Comment (tdonohue) [inline src/config/lang-config.interface.ts]: Please add a comment/TypeDocs to describe this new interface.

Comment (bram-atmire) [reply src/config/lang-config.interface.ts]: done

**Comment by bram-atmire** [comment]
@tdonohue added the requested typedoc
@artlowel reconciled the language switch with the new banner design. I thought the globe from Lotte was nice, but I hated the ngbDropdown caret next to it. Removed it using the recommendation here:
https://stackoverflow.com/questions/42891931/ngbdropdown-remove-dropdown-arrow

But you told me before the module specific .scss is not often the right place to put these css tweaks in. In this particular case, I hope it is, because I don't want to affect the behaviour of ngbDropdown in other places.

**Comment by tdonohue** [comment]
@bram-atmire : Thanks for your hard work here. I gave this another test / review today and it looks great and works great. As it was already reviewed positively by @artlowel, I'll merge it!

**Pull request merged**